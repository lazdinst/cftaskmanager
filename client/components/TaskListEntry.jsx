import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { deleteTask } from '../store/actions/app';
import { Button, Card, Label } from 'semantic-ui-react'

class TaskListEntry extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <Card fluid>
        <Card.Content>
          <Button
            basic
            color='red'
            icon='trash outline'
            floated='right' 
            size='small'
            onClick={()=>{this.props.deleteTask(this.props.task.id)}}
          />
          <Card.Header>
            {this.props.task.name || 'TASK_NAME'}
          </Card.Header>
          <Card.Meta>
            <span>
              Task id: {this.props.task.id}
              Task Number: {this.props.taskNum || 'TASK_NUM'}
            </span>
            <span className='date'>
              {this.props.task.createdAt || 'CREATED_AT'}
            </span>
        </Card.Meta>
        </Card.Content>
      </Card>
    );
  }
} 

const mapStateToProps = state => ({
  tasks: state.app.tasks,
});

const mapDispatchToProps = dispatch => bindActionCreators({
  deleteTask
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(TaskListEntry);