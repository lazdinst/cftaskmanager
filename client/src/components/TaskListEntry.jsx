import React from 'react';
import { Button, Card } from 'semantic-ui-react'

const TaskListEntry = (props) => {
  return (
    <Card fluid>
      <Card.Content>
        <Button floated='right' icon='trash outline' size='big' />
        <Card.Header>
          {props.task.name}
        </Card.Header>
      </Card.Content>
    </Card>
  );
}

export default TaskListEntry;