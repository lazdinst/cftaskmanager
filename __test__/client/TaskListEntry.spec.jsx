import React from 'react'
import { shallow, mount } from 'enzyme';
import TaskListEntry from '../../client/components/TaskListEntry.jsx'
import toJson from 'enzyme-to-json';

xdescribe('<TaskListEntry />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<TaskListEntry />);
  });

  it('should render', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
