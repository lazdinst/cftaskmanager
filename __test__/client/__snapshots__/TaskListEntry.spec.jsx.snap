// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TaskListEntry /> should render 1`] = `undefined`;
